### PR TITLE
AWS key-value store configuration

### PR DESCRIPTION
**Pull request by itowlson**
+ @ogghead (can't add you as a reviewer but would appreciate any comments you have - thanks!)

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [ ] The `title`, `template`, and `date` are all set
- [ ] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [ ] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep $'\r' | wc -l` and expect 0 as a result)
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [ ] Headings are using Title Case
- [ ] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors
- [ ] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/spin/v3/dynamic-configuration.md
Comment (ogghead) [inline content/spin/v3/dynamic-configuration.md:274]: It might be worth noting that setting this true will double your read capacity unit (RCU) usage -- generally doubling costs. Though perhaps users setting this would already be familiar with the extra cost that consistent reads entail

File: content/spin/v3/dynamic-configuration.md
Comment (ogghead) [inline content/spin/v3/dynamic-configuration.md:277]: Possibly worth also noting the possibility of passing `token` alongside these -- it comes from creating short lived credentials for a user/role using e.g. STS

File: content/spin/v3/dynamic-configuration.md
Comment (itowlson) [reply content/spin/v3/dynamic-configuration.md]: Added note with link to the AWS consistent reads page

File: content/spin/v3/dynamic-configuration.md
Comment (ogghead) [inline content/spin/v3/dynamic-configuration.md:277]: Apologies, the AWS credentials stuff is a bit funky. One must provide  `access_key` and `secret_key`, as well as potentially `token`. Just one small change

For short-lived credentials you can instead provide `token`.

to

For short-lived credentials you can additionally provide `token` from e.g. [AWS STS getSessionToken](https://docs.aws.amazon.com/STS/latest/APIReference/API_GetSessionToken.html)

**Review by ogghead — CHANGES_REQUESTED** [review]
Just one tiny change for clarity, thank you for creating this!

File: content/spin/v3/dynamic-configuration.md
Comment (itowlson) [reply content/spin/v3/dynamic-configuration.md]: Aha, thanks.  I also added a link to the CLI wrapper.

**Review by ogghead — APPROVED** [review]
lgtm! Thank you for adding this!

**Review by karthik2804 — APPROVED** [review]

File: content/spin/v3/dynamic-configuration.md
Comment (kate-goldenring) [inline content/spin/v3/dynamic-configuration.md:279]: Since the link is embedded, it may not be that clear where it is. Maybe we are redundant about the link:
```suggestion
By default, the DynamoDB backend uses eventually consistent reads. The `consistent_read` option turns on [strongly consistent reads](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadConsistency.html). This ensures reads are up-to-date with writes, at an increased cost. See the [DynamoDB documentation](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadConsistency.html) for more information.
```

**Review by kate-goldenring — APPROVED** [review]

**Comment by ogghead** [comment]
Does this PR need any additional work @itowlson? Seems like it failed build due to unrelated changes

**Comment by itowlson** [comment]
@ogghead Yeah those links are fixed on main and I will force bypass them when the time comes.  I don't believe any more work is needed; it remains open because we hold off merging feature content until the feature is in a release (because merging auto publishes), and I believe AWS KV didn't quite make it into 3.0.

**Comment by ogghead** [comment]
> it remains open because we hold off merging feature content until the feature is in a release (because merging auto publishes), and I believe AWS KV didn't quite make it into 3.0.
> 

That makes sense, thank you for the context!

**Pull request merged**
